### PR TITLE
refactor: Add logs and time measurements for different code blocks

### PR DESCRIPTION
**Pull request by muhammad-ammar**
**Merge checklist:**
- [ ] Any new requirements are in the right place (do **not** manually modify the `requirements/*.txt` files)
    - `base.in` if needed in production but edx-analytics-data-api doesn't install it
    - `test-master.in` if edx-analytics-data-api pins it, with a matching version
    - `make upgrade && make requirements` have been run to regenerate requirements
- [ ] `make static` has been run to update webpack bundling if any static content was updated
- [ ] `./manage.py makemigrations` has been run
    - Checkout the [Database Migration](https://openedx.atlassian.net/wiki/spaces/AC/pages/23003228/Everything+About+Database+Migrations) Confluence page for helpful tips on creating migrations.
    - *Note*: This **must** be run if you modified any models.
      - It may or may not make a migration depending on exactly what you modified, but it should still be run.
    - This should be run from either a venv with all the edx-analytics-data-api requirements installed or if you checked out edx-enterprise-data into the src directory used by edx-analytics-data-api, you can run this command through an edx-analytics-data-api shell.
        - It would be `./manage.py makemigrations` in the shell.
- [ ] [Version](https://github.com/openedx/edx-enterprise-data/blob/master/enterprise/__init__.py) bumped
- [ ] [Changelog](https://github.com/openedx/edx-enterprise-data/blob/master/CHANGELOG.rst) record added
- [ ] Translations updated (see docs/internationalization.rst but also this isn't blocking for merge atm)

**Post merge:**
- [ ] Tag pushed and a new [version](https://github.com/openedx/edx-enterprise-data/releases) released
    - *Note*: Assets will be added automatically. You just need to provide a tag (should match your version number) and title and description.
- [ ] After versioned build finishes in [Travis](https://travis-ci.org/github/edx/edx-enterprise-data), verify version has been pushed to [PyPI](https://pypi.org/project/edx-enterprise-data/)
    - Each step in the release build has a condition flag that checks if the rest of the steps are done and if so will deploy to PyPi.
    (so basically once your build finishes, after maybe a minute you should see the new version in PyPi automatically (on refresh))
- [ ] PR created in [edx-analytics-data-api](https://github.com/openedx/edx-analytics-data-api) to upgrade dependencies (including edx-enterprise-data)
    - This **must** be done after the version is visible in PyPi as `make upgrade` in edx-analytics-data-api will look for the latest version in PyPi.
    - Note: the edx-enterprise-data constraint in edx-analytics-data-api **must** also be bumped to the latest version in PyPi.

### CODE REVIEW

File: enterprise_data/api/v1/views/analytics_enrollments.py [hunk 1]
```diff
@@ -112,49 +129,55 @@ def get(self, request, enterprise_uuid):  # lint-amnesty, pylint: disable=incons
...
-                "top_subjects_by_enrollments": self.construct_top_subjects_by_enrollments(
-                    enrollments_df.copy(),
-                    start_date,
-                    end_date,
-                ),
-            }
+            with timer('construct_enrollment_all_stats'):
+                data = {
+                    "enrollments_over_time": self.construct_enrollments_over_time(
+                        enrollments_df.copy(),
```
Comment (saleem-latif) [inline enterprise_data/api/v1/views/analytics_enrollments.py:138]: I think we should remove calls to `copy` here.

**Review by jajjibhai008 — APPROVED** [review]
LGTM 👍

File: enterprise_data/utils.py [hunk 2]
```diff
@@ -68,6 +69,21 @@ def wrapper(*args, **kwargs):
...
+    """
+    Context manager to measure the time taken by a block of code.
+
+    Arguments:
+        prefix (str): The prefix to print in the log.
+    """
+    start = time.time()
+    yield
+    end = time.time()
+    difference = end - start
```
Comment (saleem-latif) [inline enterprise_data/utils.py:83]: nit: We can remove `end` and directly call `time.time()  - start`

**Review by saleem-latif — APPROVED** [review]

File: enterprise_data/api/v1/views/analytics_enrollments.py (hunk 1 above)
Comment (muhammad-ammar) [reply enterprise_data/api/v1/views/analytics_enrollments.py:138]: copy is needed here because every function mutates the original data frame.

**Pull request merged**